### PR TITLE
openssl: add version `3.3.1`

### PR DESCRIPTION
**Pull request by gegles**
### Summary
Changes to recipe:  **openssl/3.3.1**

#### Details

---
- [x] Read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md)
- [x] Checked that this PR is not a duplicate: [list of PRs by recipe](https://github.com/conan-io/conan-center-index/discussions/24240)
- [x] Tested locally with at least one configuration using a recent version of Conan

### CODE REVIEW

**Comment by jcar87** [comment]
Working on a fix for this on Windows... bear with us :D

File: recipes/openssl/3.x.x/conanfile.py [hunk 1]
```diff
@@ -515,6 +516,10 @@ def _make(self):
                 # When `--prefix=/`, the scripts derive `\` without escaping, which
                 # causes issues on Windows
                 replace_in_file(self, "Makefile", "INSTALLTOP_dir=\\", "INSTALLTOP_dir=\\\\")
+                if Version(self.version) >= "3.3.0":
+                    # replace backslashes in paths with forward slashes
+                    mkinstallvars_pl = os.path.join(self.source_folder, "util", "mkinstallvars.pl")
+                    replace_in_file(self, mkinstallvars_pl, "$ENV{$k} = $v;", """$v =~ s|\\\\|/|g; $ENV{$k} = $v;""")
```
Comment (jcar87) [inline recipes/openssl/3.x.x/conanfile.py:522]: This fixes the following error:

```
Microsoft (R) Program Maintenance Utility Version 14.29.30148.0
Copyright (C) Microsoft Corporation.  All rights reserved.

	"C:/J/workspace/prod-v2/bsr/65821/bcebd/p/straw898fb56546adc/p/bin/perl.exe" "util\mkinstallvars.pl" PREFIX=. BINDIR=apps LIBDIR= INCLUDEDIR=include APPLINKDIR=ms ENGINESDIR=engines MODULESDIR=providers "VERSION=3.3.1" "LDLIBS=C:/J/workspace/prod-v2/bsr/65821/bcebd/p/zlibbe5a5af71d5cc/p/lib/zlib.lib ws2_32.lib gdi32.lib advapi32.lib crypt32.lib user32.lib " > builddata.pm
	"C:/J/workspace/prod-v2/bsr/65821/bcebd/p/straw898fb56546adc/p/bin/perl.exe" "-I." "-Mconfigdata" "-Mbuilddata" "util\dofile.pl" "-omakefile" "exporters\cmake\OpenSSLConfig.cmake.in" > OpenSSLConfig.cmake
	"C:/J/workspace/prod-v2/bsr/65821/bcebd/p/straw898fb56546adc/p/bin/perl.exe" "-I." "-Mconfigdata" "-Mconfigdata" "-Mbuilddata" "util\dofile.pl" "-omakefile" "exporters\cmake\OpenSSLConfigVersion.cmake.in" > OpenSSLConfigVersion.cmake
	"C:/J/workspace/prod-v2/bsr/65821/bcebd/p/straw898fb56546adc/p/bin/perl.exe" "-I." "-Iutil\perl" "-Mconfigdata" "-MOpenSSL::paramnames" "util\dofile.pl" "-omakefile" "crypto\params_idx.c.in" > crypto\params_idx.c
	"C:/J/workspace/prod-v2/bsr/65821/bcebd/p/straw898fb56546adc/p/bin/perl.exe" "util\mkinstallvars.pl" "PREFIX=\\" BINDIR=bin "LIBDIR=lib" INCLUDEDIR=include APPLINKDIR=include/openssl "ENGINESDIR=\lib\engines-3" "MODULESDIR=\lib\ossl-modules" "PKGCONFIGDIR=" "CMAKECONFIGDIR=\\\lib\cmake\OpenSSL" "LDLIBS=C:/J/workspace/prod-v2/bsr/65821/bcebd/p/zlibbe5a5af71d5cc/p/lib/zlib.lib ws2_32.lib gdi32.lib advapi32.lib crypt32.lib user32.lib " "VERSION=3.3.1" > installdata.pm
	"C:/J/workspace/prod-v2/bsr/65821/bcebd/p/straw898fb56546adc/p/bin/perl.exe" "-I." "-Mconfigdata" "-Minstalldata" "util\dofile.pl" "-omakefile" "exporters\cmake\OpenSSLConfig.cmake.in" > exporters\OpenSSLConfig.cmake
Bareword found where operator expected at installdata.pm line 19, near "our $BINDIR             = 'C"
  (Might be a runaway multi-line '' string starting on line 18)
	(Do you need to predeclare our?)
```

The `"PREFIX=/"` when calling `mkinstallvars.pl` ends up generating a file where `/` is resolved as `C:\` and then the backslashes are not escaped,  resulting in a malformed generated `installdata.pm`. So this just replaces the slashes with forward ones. Unsure why it worked before - we are also not writing in those locations anyway (that is, we shouldn't end up with files written in C:/)

**Comment by jcar87** [comment]
Note to Conan team reviewers: if/when this is merged, it will cause the unconstrained solver to use OpenSSL 3.3.1 where it may previously resolve to 3.2.x - this could cause missing package binaries in subsequent PRs - so we need to publish new packages for anything that may have previously referenced openssl 3.2 - this needs to be coordinated _before_ merging this PR.

File: recipes/openssl/3.x.x/conanfile.py (hunk 1 above)
Comment (gegles) [reply recipes/openssl/3.x.x/conanfile.py:522]: Thank you so much @jcar87!

**Review by AbrilRBS — APPROVED** [review]
Thanks! We're now in the process of adding this new version to the repository ensuring there won't be missing binaries :)

**Review by ErniGH — APPROVED** [review]

**Comment by jcar87** [comment]
Whist the system is in scheduled maintenance today - we will proceed to the build all packages that would have their package ID altered by the `3.2.x` -> `3.3.x` update.

These are quite a bit. We will prioritise these to prevent "missing binaries" in further pull request CI runs, as well as disruption to users who are not using lockfiles (although for the latter we are constrained by how fast CI can process these).


```
[["kcov/38", "kcov/40", "kcov/42", "kmod/29", "kmod/30", "kmod/31", "mold/1.4.2", "mold/1.5.1", "mold/1.7.1", "mold/1.8.0", "mold/1.11.0", "mold/2.0.0", "mold/2.4.0", "mold/2.31.0", "mold/2.32.0", "mold/2.32.1", "openssh/9.1p1", "openssh/9.6p1", "pistache/cci.20201127", "rsync/3.2.7"], ["civetweb/1.13", "civetweb/1.14", "civetweb/1.15", "libmysqlclient/8.0.25", "proj/6.3.1", "pulseaudio/17.0"], ["amqp-cpp/4.1.7", "amqp-cpp/4.2.1", "amqp-cpp/4.3.24", "amqp-cpp/4.3.26", "civetweb/1.16", "libcurl/8.9.1", "libevent/2.1.12", "aws-c-cal/0.5.13", "s2n/1.3.55", "aws-c-cal/0.5.12", "aws-c-cal/0.5.17", "aws-c-cal/0.5.19", "aws-c-cal/0.5.20", "aws-c-cal/0.6.1", "aws-c-cal/0.6.9", "aws-c-event-stream/0.1.5", "s2n/1.4.1", "beauty/1.0.2", "beauty/1.0.3", "beauty/1.0.4", "c-client/2007f", "caf/0.18.0", "caf/0.18.2", "caf/0.18.5", "caf/0.18.6", "caf/0.19.2", "caf/0.19.3", "caf/0.19.4", "caf/1.0.0", "capnproto/0.7.0", "capnproto/0.8.0", "capnproto/0.9.1", "capnproto/0.10.0", "capnproto/0.10.1", "capnproto/0.10.3", "capnproto/0.10.4", "capnproto/1.0.0", "capnproto/1.0.1", "capnproto/1.0.2", "cassandra-cpp-driver/2.15.3", "cassandra-cpp-driver/2.16.2", "cassandra-cpp-driver/2.17.0", "cassandra-cpp-driver/2.17.1", "cnats/3.8.0", "cose-c/cci.20200430", "cppserver/1.0.0.0", "cppserver/1.0.1.0", "cppserver/1.0.2.0", "cyrus-sasl/2.1.27", "cyrus-sasl/2.1.28", "dcmtk/3.6.7", "dcmtk/3.6.8", "trantor/1.5.19", "libmysqlclient/8.1.0", "gamenetworkingsockets/1.4.1", "gdcm/3.0.9", "gdcm/3.0.20", "gdcm/3.0.21", "gdcm/3.0.22", "gdcm/3.0.23", "gdcm/3.0.24", "grpc/1.50.0", "grpc/1.50.1", "gsoap/2.8.129", "gsoap/2.8.132", "gsoap/2.8.134", "ldns/1.8.1", "ldns/1.8.2", "libcoap/4.3.0", "libcoap/4.3.3", "libcoap/cci.20200424", "libcoro/0.7", "libcoro/0.8", "libcoro/0.9", "libcoro/0.10", "libcoro/0.11.1", "libcoro/0.12", "libcoro/0.12.1", "libcurl/7.88.1", "libcurl/8.0.1", "libcurl/8.1.2", "libcurl/8.2.1", "libcurl/8.4.0", "libcurl/8.5.0", "libcurl/8.6.0", "libcurl/8.8.0", "libevent/2.1.11", "libftp/0.4.1", "libftp/0.5.0", "libftp/0.5.1", "libftp/1.1.0", "libssh2/1.11.0", "libmysqlclient/8.0.30", "libmysqlclient/8.0.31", "libmysqlclient/8.0.34", "librhash/1.3.9", "librhash/1.4.2", "librhash/1.4.4", "libssh/0.10.6", "libssh2/1.8.0", "libssh2/1.8.2", "libssh2/1.9.0", "libssh2/1.10.0", "libtins/4.2", "libtins/4.3", "libtins/4.4", "libtins/4.5", "libtorrent/1.2.3", "libtorrent/2.0.1", "libtorrent/2.0.8", "libtorrent/2.0.9", "libtorrent/2.0.10", "libwebsockets/4.2.0", "libwebsockets/4.2.1", "libwebsockets/4.3.0", "libwebsockets/4.3.2", "libwebsockets/4.3.3", "libzip/1.7.3", "libzip/1.8.0", "libzip/1.9.2", "libzip/1.10.1", "mailio/0.20.0", "mailio/0.21.0", "mailio/0.22.0", "mailio/0.23.0", "mgclient/1.4.2", "minizip-ng/3.0.7", "minizip-ng/3.0.8", "minizip-ng/3.0.9", "minizip-ng/3.0.10", "minizip-ng/4.0.0", "minizip-ng/4.0.1", "minizip-ng/4.0.2", "minizip-ng/4.0.3", "minizip-ng/4.0.4", "minizip-ng/4.0.5", "minizip-ng/4.0.6", "mongo-c-driver/1.23.5", "mongo-c-driver/1.24.3", "mongo-c-driver/1.26.2", "mongo-c-driver/1.27.1", "mongo-c-driver/1.27.2", "mongo-c-driver/1.27.3", "mongo-c-driver/1.27.4", "mosquitto/2.0.3", "mosquitto/2.0.5", "mosquitto/2.0.7", "mosquitto/2.0.10", "mosquitto/2.0.11", "mosquitto/2.0.12", "mosquitto/2.0.14", "mosquitto/2.0.15", "mosquitto/2.0.18", "msix/1.7", "mysql-concpp/9.0.0", "nuraft/2.0.0", "nuraft/2.1.0", "oatpp-openssl/1.2.0", "oatpp-openssl/1.2.5", "oatpp-openssl/1.3.0", "octo-encryption-cpp/1.1.0", "ohpipeline/1.139.1000", "ohpipeline/1.155.1133", "opencolorio/1.1.1", "opencolorio/2.1.0", "opencv/2.4.13.7", "opencv/3.4.17", "opencv/3.4.20", "opusfile/0.11", "opusfile/0.12", "paho-mqtt-c/1.3.4", "paho-mqtt-c/1.3.5", "paho-mqtt-c/1.3.6", "paho-mqtt-c/1.3.8", "paho-mqtt-c/1.3.9", "paho-mqtt-c/1.3.10", "paho-mqtt-c/1.3.11", "paho-mqtt-c/1.3.12", "paho-mqtt-c/1.3.13", "qpdf/11.1.1", "restbed/4.8", "s2geometry/0.10.0", "s2geometry/0.11.1", "s2n/1.0.11", "s2n/1.1.0", "s2n/1.1.1", "s2n/1.2.0", "s2n/1.3.9", "s2n/1.3.15", "s2n/1.3.31", "s2n/1.3.49", "s2n/1.3.50", "s2n/1.3.52", "s2n/1.3.56", "s2n/1.4.0", "s2n/1.4.13", "s2n/1.4.16", "serf/1.3.9", "sqlcipher/4.3.0", "sqlcipher/4.4.0", "sqlcipher/4.4.2", "sqlcipher/4.4.3", "sqlcipher/4.5.0", "sqlcipher/4.5.1", "sqlcipher/4.5.6", "sqlcipher/4.5.7", "sqlcipher/4.6.0", "srt/1.4.4", "srt/1.5.3", "trantor/1.5.13", "trantor/1.5.14", "trantor/1.5.15", "trantor/1.5.16", "trantor/1.5.20", "xmlsec/1.2.30", "xmlsec/1.2.31", "xmlsec/1.2.32", "xmlsec/1.2.33", "xmlsec/1.2.38", "xmlsec/1.3.2", "xmlsec/1.3.3", "xmlsec/1.3.4"], ["prometheus-cpp/1.1.0", "aws-c-io/0.10.20", "aws-c-io/0.13.4", "aws-c-io/0.13.35", "aws-c-io/0.10.13", "aws-c-io/0.11.2", "aws-c-io/0.13.32", "aws-c-io/0.14.1", "aws-lambda-cpp/0.2.8", "aws-lambda-cpp/0.2.9", "aws-lambda-cpp/0.2.10", "azure-sdk-for-cpp/1.11.3", "bmx/1.2", "bmx/cci.20240517", "cpprestsdk/2.10.15", "cpprestsdk/2.10.16", "cpprestsdk/2.10.17", "cpprestsdk/2.10.18", "cpprestsdk/2.10.19", "cpr/1.7.2", "cpr/1.8.1", "cpr/1.9.3", "cpr/1.10.4", "cpr/1.10.5", "crashpad/cci.20220219", "date/2.4.1", "date/3.0.0", "date/3.0.1", "dd-opentracing-cpp/1.3.0", "dd-opentracing-cpp/1.3.7", "drogon/1.8.7", "drogon/1.9.0", "drogon/1.9.1", "drogon/1.9.2", "drogon/1.9.3", "drogon/1.9.4", "drogon/1.9.5", "drogon/1.9.6", "poco/1.12.4", "foxglove-websocket/1.0.0", "foxglove-websocket/1.1.0", "foxglove-websocket/1.2.0", "foxglove-websocket/1.3.0", "proj/9.3.1", "libbigwig/0.4.7", "libgit2/1.0.1", "libgit2/1.1.1", "libgit2/1.2.0", "libgit2/1.3.0", "libgit2/1.4.3", "libgit2/1.5.0", "libgit2/1.7.2", "libgit2/1.8.1", "libmediainfo/20.03", "libmediainfo/21.03", "libmediainfo/21.09", "libmediainfo/22.03", "libsixel/1.10.3", "libvault/0.59.0", "libverto/0.3.2", "libzippp/4.0", "libzippp/5.0-1.8.0", "libzippp/6.0-1.9.2", "libzippp/6.1-1.9.2", "libzippp/7.0-1.10.1", "libzippp/7.1-1.10.1", "mariadb-connector-c/3.1.11", "mariadb-connector-c/3.1.12", "mariadb-connector-c/3.1.19", "mariadb-connector-c/3.3.3", "mariadb-connector-c/3.3.8", "mongo-cxx-driver/3.6.1", "mongo-cxx-driver/3.6.2", "mongo-cxx-driver/3.6.6", "mongo-cxx-driver/3.6.7", "mongo-cxx-driver/3.7.0", "mongo-cxx-driver/3.7.2", "mongo-cxx-driver/3.8.0", "mongo-cxx-driver/3.8.1", "mongo-cxx-driver/3.10.1", "mongo-cxx-driver/3.10.2", "mosquitto/1.6.12", "netcdf/4.7.4", "netcdf/4.8.1", "octo-keygen-cpp/1.0.0", "opencolorio/2.2.1", "opencolorio/2.3.0", "opencolorio/2.3.1", "opencolorio/2.3.2", "openldap/2.6.1", "openldap/2.6.7", "opentdf-client/1.3.10", "opentdf-client/1.4.0", "opentdf-client/1.5.0", "opentdf-client/1.5.3", "opentdf-client/1.5.4", "opentdf-client/1.5.6", "ouster_sdk/0.10.0", "ouster_sdk/0.11.0", "ouster_sdk/0.12.0", "paho-mqtt-cpp/1.2.0", "paho-mqtt-cpp/1.3.2", "paho-mqtt-cpp/1.4.0", "poco/1.11.3", "poco/1.12.0", "poco/1.12.1", "poco/1.12.2", "poco/1.12.3", "poco/1.12.5p1", "poco/1.12.5p2", "poco/1.13.0", "poco/1.13.1", "poco/1.13.2", "poco/1.13.3", "proj/7.2.1", "proj/8.2.1", "proj/9.2.1", "proj/9.3.0", "proj/9.4.1", "prometheus-cpp/0.11.0", "prometheus-cpp/0.12.1", "prometheus-cpp/0.12.3", "prometheus-cpp/1.0.0", "prometheus-cpp/1.0.1", "sentry-crashpad/0.4.18", "sentry-crashpad/0.5.4", "sentry-crashpad/0.6.3", "sentry-crashpad/0.6.4", "sentry-crashpad/0.6.5", "sentry-native/0.5.4", "sentry-native/0.6.6", "sentry-native/0.7.4", "sentry-native/0.7.5", "sentry-native/0.7.6", "sioclient/cci.20240405", "teemo/2.7", "tesseract/4.1.1", "tesseract/5.0.0", "tesseract/5.1.0", "tesseract/5.2.0", "tesseract/5.3.0", "tesseract/5.3.3", "tesseract/5.3.4", "tesseract/5.4.0", "tesseract/5.4.1", "tgbot/1.3", "tgbot/1.5", "tgbot/1.7.2", "thrift/0.14.1", "thrift/0.14.2", "thrift/0.15.0", "thrift/0.16.0", "thrift/0.17.0", "thrift/0.18.1", "thrift/0.20.0", "tixi3/3.3.0", "very-simple-smtps/1.0.0", "wt/4.6.0", "wt/4.6.2", "wt/4.7.1", "wt/4.8.0", "wt/4.9.1", "wt/4.10.0", "wt/4.10.1", "zoe/3.0", "zoe/3.1"], ["pulseaudio/14.2", "asyncly/cci.20230420", "avahi/0.8", "aws-c-http/0.6.13", "aws-c-http/0.6.22", "aws-c-event-stream/0.2.7", "aws-c-event-stream/0.2.15", "aws-c-event-stream/0.2.20", "aws-c-event-stream/0.3.1", "aws-c-event-stream/0.3.2", "aws-c-http/0.6.5", "aws-c-http/0.6.7", "aws-c-http/0.6.10", "aws-c-http/0.7.14", "aws-c-http/0.8.1", "easyhttpcpp/2.1.0", "grpc/1.54.3", "libgeotiff/1.7.1", "grpc/1.65.0", "influxdb-cxx/0.7.2", "libgeotiff/1.5.1", "libgeotiff/1.6.0", "libgeotiff/1.7.0", "libgphoto2/2.5.27", "libgphoto2/2.5.31", "libnice/0.1.21", "libspatialite/5.0.1", "libspatialite/5.1.0", "pistache/0.0.5", "pistache/cci.20240107", "pulseaudio/13.0", "pulseaudio/14.0", "unleash-client-cpp/1.3.0"], ["ffmpeg/4.4.4", "ffmpeg/6.1", "aws-c-auth/0.6.0", "aws-c-auth/0.6.4", "aws-c-auth/0.6.8", "aws-c-auth/0.6.11", "aws-c-auth/0.6.17", "aws-c-mqtt/0.7.6", "aws-c-mqtt/0.7.9", "aws-c-mqtt/0.7.10", "aws-c-mqtt/0.7.12", "aws-sdk-cpp/1.8.130", "cpython/3.8.19", "cpython/3.9.7", "cpython/3.9.19", "cpython/3.10.14", "cpython/3.11.9", "cpython/3.12.2", "etcd-cpp-apiv3/0.15.4", "ffmpeg/5.0.3", "ffmpeg/5.1.3", "ffmpeg/6.0.1", "ffmpeg/6.1.1", "ffmpeg/7.0.1", "gdal/3.4.1", "gdal/3.4.3", "gdal/3.5.3", "gdal/3.7.3", "gdal/3.8.3", "google-cloud-cpp/2.12.0", "google-cloud-cpp/2.15.1", "google-cloud-cpp/2.19.0", "libmikmod/3.3.11.1", "qt/5.15.14", "nmos-cpp/cci.20220120", "nmos-cpp/cci.20220208", "nmos-cpp/cci.20220428", "nmos-cpp/cci.20220620", "nmos-cpp/cci.20221203", "nmos-cpp/cci.20240223", "opentelemetry-cpp/1.12.0", "opentelemetry-cpp/1.14.2", "qt/5.15.7", "qt/5.15.9", "qt/5.15.11", "qt/5.15.13", "qt/6.5.3", "qt/6.6.1", "qt/6.6.3", "qt/6.7.1", "sdl/2.0.20", "sdl/2.24.0", "sdl/2.24.1", "sdl/2.26.0", "sdl/2.26.1", "sdl/2.26.5", "sdl/2.28.2", "sdl/2.28.3", "sdl/2.28.5", "sdl/2.30.1", "sdl/2.30.2", "sdl/2.30.3", "sdl/2.30.4", "sdl/2.30.5", "sdl/2.30.6", "sqlpp11-connector-sqlite3/0.29", "sqlpp11-connector-sqlite3/0.30"], ["opencv/4.5.5", "avcpp/2.1.0", "avcpp/2.3.0", "avcpp/2.3.2", "aws-c-s3/0.1.27", "aws-c-s3/0.1.29", "aws-c-s3/0.1.37", "aws-c-s3/0.1.49", "opencv/4.9.0", "opencv/4.8.1", "mocknetworkaccessmanager/0.12.0", "opencv/4.1.2", "opencv/4.5.3", "opencv/4.10.0", "openimageio/2.4.7.1", "openimageio/2.4.17.0", "openimageio/2.5.10.1", "openimageio/2.5.12.0", "openimageio/2.5.13.0", "openimageio/2.5.13.1", "openimageio/2.5.14.0", "qarchive/2.0.1", "qarchive/2.0.2", "qarchive/2.1.1", "qarchive/2.2.3", "qarchive/2.2.4", "qarchive/2.2.6", "qarchive/2.2.8", "qcustomplot/1.3.2", "qtawesome/6.4.0", "qtxlsxwriter/0.3.0", "quazip/1.1", "quazip/1.2", "quazip/1.3", "quazip/1.4", "qwt/6.2.0", "qxlsx/1.4.3", "qxlsx/1.4.4", "qxlsx/1.4.5", "qxmpp/1.4.0", "runtimeqml/cci.20211220", "sdl_image/2.0.5", "sdl_image/2.6.3", "sdl_mixer/2.8.0", "sdl_net/2.0.1", "sdl_net/2.2.0", "sdl_ttf/2.0.15", "sdl_ttf/2.0.18", "sdl_ttf/2.20.1", "sdl_ttf/2.20.2", "sdl_ttf/2.22.0"], ["aruco/3.1.12", "aws-crt-cpp/0.14.3", "aws-crt-cpp/0.17.12", "aws-crt-cpp/0.17.23", "cctag/1.0.1", "cctag/1.0.3", "cctag/1.0.4", "imutils-cpp/1.0.1", "stella-cv-fbow/cci.20240508"], ["aws-sdk-cpp/1.9.100", "aws-sdk-cpp/1.9.234"]]
```

**Pull request merged**